pulsar/consumer_impl.go

```diff
@@ -707,6 +709,34 @@ func (c *consumer) checkMsgIDPartition(msgID MessageID) error {
 	return nil
 }
 
+func (c *consumer) hasNext() bool {
+	var wg sync.WaitGroup
+
+	wg.Add(len(c.consumers))
+
+	var hasNext uAtomic.Bool
+	for _, pc := range c.consumers {
+		pc := pc
+		go func() {
+			defer wg.Done()
+			if pc.hasNext() {
+				hasNext.Store(true)
```

Comment:
if a `pc` has the next message, we can stop the wait early?

Reply:
I had considered this before. However, I didn't implement it then because it required introducing a channel and context, which could add complexity and risks but without significant benefits. Anyway, I've just added this implementation and have carefully checked to ensure there won't be any panic.